torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -204,6 +205,21 @@ def get_profiling_event(event_name, profiler, dedup_gpu_user_annotation=False):
         )
     ]
 
+def get_profiler_nccl_meta(prof):
+    """Torch profiler includes nccl metadata in an inserted operator called "record_param_comms"
+    We will need to test metadata obtained from profiler here"""
+    tf = tempfile.NamedTemporaryFile(
+        mode="w+t", suffix=".json", delete=False
+    )
```

Comment:
nit: Does profiler have an in-program export method instead of saving and loading from file? 

nit: if via file is a must, shall we consider deleting the file at the end?

Reply:
@kwen2501 yes it does need a file. Usually it is easier to have the file around after the test to enable debugging :) The file is anyway saved in temp so not a big issue

Reply:
Hmm, I would not assume the CI machine would clean things up automatically. Would appreciate it if the test could be made self sufficient.
And CI tests are not for debugging purpose mainly.